### PR TITLE
Notif suggestion (settings switch)

### PR DESCRIPTION
**Pull request by alanlanglois**
## Detailed purpose of the PR

Adding a new switch in settings for recommandation notifications.

### Testing strategy

- [ ] I ensured my code is tested: 
- [ ] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

File: src/redux/reducers/notifications/suggestion.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export type State = {
+  type: NotificationType;
+  label: string;
+  isActive: boolean;
+};
+
+const initialState: State = {
+  type: NOTIFICATION_TYPE.SUGGESTION,
+  label: 'This course might be for you', //TODO add translations
+  isActive: Platform.OS === 'android',
```
Comment (Fausto95) [inline src/redux/reducers/notifications/suggestion.ts:17]:
```suggestion
  isActive: true,
```

File: src/screens/settings.tsx [hunk 2]
```diff
@@ -25,11 +26,15 @@ const styles = StyleSheet.create({
 export interface ConnectedStateProps {
   canReceiveNotifications: boolean;
   currentNotificationsPermission: PermissionStatus;
-  notificationsSettings: Record<'finishCourse', Pick<NotificationsState, 'finishCourse'>>;
+  notificationsSettings: Record<
```
Comment (Fausto95) [inline src/screens/settings.tsx:29]:
```ts
notificationsSettings: {
    'finish-course': Pick<NotificationsState, 'finishCourse'>,
    suggestion: Pick<NotificationsState, 'suggestion'>
}
```

File: src/screens/settings.tsx [hunk 3]
```diff
@@ -65,6 +72,10 @@ const SettingsScreen = ({
           _toggleFinishCourseNotification();
           break;
         }
+        case NOTIFICATION_TYPE.SUGGESTION: {
+          _toggleSuggestionNotification();
+          break;
+        }
```
Comment (Fausto95) [inline src/screens/settings.tsx:78]: I think we could have a single toggle function that takes in a type and does the actual toggle.

**Review by Fausto95 — APPROVED** [review]

**Review by Fausto95 — CHANGES_REQUESTED** [review]
See comments

File: src/redux/reducers/notifications/suggestion.ts (hunk 1 above)
Comment (Fausto95) [reply src/redux/reducers/notifications/suggestion.ts:17]: Don't take this comment is consideration,
But we should somewhere set this to true if the iOS user has already granted notifications permissions to us

File: src/screens/settings.tsx (hunk 3 above)
Comment (alanlanglois) [reply src/screens/settings.tsx:78]: Yep, you're right it might be easier to maintain.

File: src/screens/settings.tsx [hunk 4]
```diff
@@ -43,10 +38,20 @@ interface Props
     ConnectedDispatchProps,
     WithPermissionsProps {}
 
+export const transformNotificationSettings = (settings: NotificationsSettingsState) => {
```
Comment (Fausto95) [inline src/screens/settings.tsx:41]: For some reason doing this transformation screws up the order of the elements, hence the sorting

File: src/redux/actions/notifications/scheduled-notifications.test.ts [hunk 5]
```diff
@@ -25,10 +25,12 @@ const createStore = (isActive: boolean, scheduledNotifications?: ScheduledNotifi
       },
     },
     notifications: {
-      finishCourse: {
-        type: 'finish-course',
-        label: 'Finish course',
-        isActive: isActive,
+      settings: {
+        'finish-course': {
+          type: 'finish-course',
```
Comment (alanlanglois) [inline src/redux/actions/notifications/scheduled-notifications.test.ts:30]: remove that

**Pull request merged**